interpreter/go/go.go

```diff
@@ -0,0 +1,164 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package golang // import "go.opentelemetry.io/ebpf-profiler/interpreter/go"
+
+/*
+#cgo CFLAGS: -g -Wall
+#include "../../rust-crates/symblib-capi/c/symblib.h"
+#include <stdlib.h>
+*/
+import "C"
+
+import (
+	"errors"
+	"fmt"
+	"hash/fnv"
+	"sync/atomic"
+	"unsafe"
+
+	"go.opentelemetry.io/ebpf-profiler/host"
+	"go.opentelemetry.io/ebpf-profiler/interpreter"
+	"go.opentelemetry.io/ebpf-profiler/libpf"
+	"go.opentelemetry.io/ebpf-profiler/metrics"
+	"go.opentelemetry.io/ebpf-profiler/remotememory"
+	"go.opentelemetry.io/ebpf-profiler/reporter"
+	"go.opentelemetry.io/ebpf-profiler/successfailurecounter"
+)
+
+var (
+	// compiler check to make sure the needed interfaces are satisfied
+	_ interpreter.Data     = &goData{}
+	_ interpreter.Instance = &goInstance{}
+)
+
+type goData struct {
+	goExecutable *C.SymblibPointResolver
+}
+
+type goInstance struct {
+	interpreter.InstanceStubs
+
+	// Go symbolization metrics
+	successCount atomic.Uint64
+	failCount    atomic.Uint64
+
+	d *goData
+}
+
+func Loader(_ interpreter.EbpfHandler, info *interpreter.LoaderInfo) (
+	interpreter.Data, error) {
+	ef, err := info.GetELF()
+	if err != nil {
+		return nil, err
+	}
+	if !ef.IsGolang() {
+		return nil, nil
+	}
+
+	exec, err := info.ExtractAsFile()
+	if err != nil {
+		return nil, err
+	}
+
+	executablePath := C.CString(exec)
+	defer C.free(unsafe.Pointer(executablePath))
+
+	gd := &goData{}
+
+	//nolint:gocritic
+	status := C.symblib_goruntime_new(executablePath, &gd.goExecutable)
```

Comment:
Now that this is in `Loader` this will leak memory. There has been a deficiency in the API that there is no corresponding `interpreter.Data.Unload` method. I created #426 to introduce this. Perhaps you could rebase on top of it, and add the corresponding free in `Unload`?

Reply:
This has been done with https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/408/commits/733570fad5bfb5407d82b8253327f756cfc63adb. Therefore resolving.